sktime/forecasting/reconcile.py

```diff
@@ -0,0 +1,355 @@
+    _parent_child_df,
+)
+
+
+class ReconcilerForecaster(BaseForecaster):
+    """Hierarchical reconcilation forecaster.
+
+    Reconciliation is applied to make the forecasts in a hierarchy of
+    time-series sum together appropriately.
+
+    Please refer to [1]_ for further information.
+
+    Parameters
+    ----------
+    forecaster : estimator
+        Estimator to generate base forecasts which are then reconciled
+    method : {"mint_cov", "mint_shrink", "wls_var"}, default="mint_cov"
+        The reconciliation approach applied to the forecasts based on
+            "mint_cov" - sample covariance
+            "mint_shrink" - covariance with shrinkage
+            "wls_var" - weighted least squares (variance)
+
+    References
+    ----------
+    .. [1] https://otexts.com/fpp3/hierarchical.html
+    """
+
+    _required_parameters = ["forecaster"]
+
+    _tags = {
+        "scitype:y": "univariate",  # which y are fine? univariate/multivariate/both
+        "ignores-exogeneous-X": False,  # does estimator ignore the exogeneous X?
+        "handles-missing-data": False,  # can estimator handle missing data?
+        "y_inner_mtype": [
+            "pd.DataFrame",
+            "pd.Series",
+            "pd-multiindex",
+            "pd_multiindex_hier",
+        ],
+        "X_inner_mtype": [
+            "pd.DataFrame",
+            "pd.Series",
+            "pd-multiindex",
+            "pd_multiindex_hier",
+        ],  # which types do _fit, _predict, assume for X?
+        "requires-fh-in-fit": False,  # is forecasting horizon already required in fit?
+        "X-y-must-have-same-index": False,  # can estimator handle different X/y index?
+        "enforce_index_type": None,  # index type that needs to be enforced in X/y
+        "capability:pred_int": False,  # does forecaster implement proba forecasts?
+        "fit_is_empty": False,
+    }
+
+    METHOD_LIST = ["mint_cov", "mint_shrink", "wls_var"]
+
+    def __init__(self, forecaster, method="mint_cov"):
+
+        self.forecaster = forecaster
+        self.method = method
+
+        super(ReconcilerForecaster, self).__init__()
+
+    def _add_totals(self, y):
+        """Add total levels to y, using Aggregate."""
+        from sktime.transformations.hierarchical.aggregate import Aggregator
+
+        return Aggregator().fit_transform(y)
+
+    def _fit(self, y, X=None, fh=None):
+        """Fit forecaster to training data.
+
+        Parameters
+        ----------
+        y : pd.DataFrame
+            Time series to which to fit the forecaster.
+        fh : ForecastingHorizon or None, optional (default=None)
+            The forecasting horizon with the steps ahead to to predict.
+        X : pd.DataFrame, default=None
+            Exogenous variables for the base forecaster
+
+        Returns
+        -------
+        self : reference to self
+        """
+        self._check_method()
+
+        # # check the length of index if not hierarchical just return forecaster
```

Comment:
well, not 100% precise, you fit the forecaster and return reference to self (which is the correct thing to do)